### PR TITLE
Tutorial/tum

### PR DESCRIPTION
**Pull request by andre-merzky**
This branch contains some documentation, example and code fixes from the Munich tutorial.  Only functional change is to add file staging to the slurm adaptor.

### CODE REVIEW

File: docs/source/usage/config.rst [hunk 1]
```diff
@@ -5,8 +5,13 @@ Configuration
 
 .. note::
 
-   SAGA has been designed as a zero-configuration library. Unless you are 
-   experiencing problems with one of the default configuration settings, there's 
+   This section is outdated!
+
```
Comment (mturilli) [inline docs/source/usage/config.rst]: We should open a ticket with medium to severe priority about updating it. SAGA is in maintenance mode so I think that outdated documentation is a greater problem than with a rapidly changing code base.

File: docs/source/usage/logger.rst [hunk 2]
```diff
@@ -17,101 +17,76 @@ the logging system for advanced use-cases.
...
-
-   app_logger       = getLogger ('application.test')
-   app_logger.level = INFO
-
-   app_logger.info ('application level log message on INFO level')
-
-
-.. automodule:: saga.utils.logger
-   :members:
+       SAGA_VERBOSE=DEBUG SAGA_LOG_TARGETS=stdout,./rs.log python mysagaprog.py
```
Comment (mturilli) [inline docs/source/usage/logger.rst]: `RADICAL_SAGA_LOG_TGT` or `SAGA_LOG_TARGETS`

File: docs/source/usage/logger.rst [hunk 3]
```diff
@@ -17,101 +17,76 @@ the logging system for advanced use-cases.
...
-
-
-.. envvar:: SAGA_LOG_TARGETS
-
-   Controls where the log messages go. Multiple concurrent locations are 
-   supported. 
-   ``SAGA_LOG_TARGETS`` expects either a single location or a comma-separated 
+   Controls where the log messages go. Multiple concurrent locations are
+   supported.
+   ``SAGA_LOG_TARGETS`` expects either a single location or a comma-separated
```
Comment (mturilli) [inline docs/source/usage/logger.rst]: `RADICAL_SAGA_LOG_TGT` or `SAGA_LOG_TARGETS`

File: docs/source/usage/logger.rst [hunk 4]
```diff
@@ -17,101 +17,76 @@ the logging system for advanced use-cases.
...
-   hierarchal structure for its log sources. Starting with the root logger
-   ``saga``, several sub loggers are defined for SAGA-internal logging events
-   (``saga.engine``) and individual middleware adaptors ``saga.adaptor.name``.
-   ``SAGA_LOG_FILTERS`` expects either a single source name or a 
-   comma-separated list of source names. Non-existing source names are ignored.
 
-   For example, if you want to see only the debug messages generated by 
-   ``saga.engine`` and a specific middleware adaptor called ``xyz`` you would 
-   set the following environment variables::
+.. envvar:: RADICAL_SAGA_LOG_TGT
```
Comment (mturilli) [inline docs/source/usage/logger.rst]: `RADICAL_SAGA_LOG_TGT` or `SAGA_LOG_TARGETS`

File: docs/source/usage/logger.rst [hunk 5]
```diff
@@ -17,101 +17,76 @@ the logging system for advanced use-cases.
...
-executing shell and evaluated by SAGA at program startup. 
+with an existing SAGA application. Environment variables are set in the
+executing shell and evaluated by SAGA at program startup.
 
-.. envvar:: SAGA_VERBOSE
+.. envvar:: RADICAL_SAGA_LOG_LVL
 
    Controls the log level. This controls the amount of output generated by the
-   logging system. ``SAGA_VERBOSE`` expects either a numeric (0-4) value or a 
+   logging system. ``SAGA_VERBOSE`` expects either a numeric (0-4) value or a
```
Comment (mturilli) [inline docs/source/usage/logger.rst]: `RADICAL_SAGA_LOG_LVL` or `SAGA_VERBOSE`

File: examples/jobs/slurmjobcontainer.py [hunk 6]
```diff
@@ -0,0 +1,117 @@
...
+__author__    = 'RADICAL @ Rutgers'
+__copyright__ = 'Copyright 2012-2013, The SAGA Project'
+__license__   = 'MIT'
+
+
+'''
+This examples shows how to run groups of jobs using the 'local' file adaptor.
+This example uses job containers for simplified and optimized bulk job handling.
+
+Job container can be used to easily model dependencies between groups of
```
Comment (mturilli) [inline examples/jobs/slurmjobcontainer.py]: I would eliminate `easily`.

File: src/radical/saga/adaptors/noop/noop_job.py [hunk 7]
```diff
@@ -0,0 +1,683 @@
+
+__author__    = "Andre Merzky"
+__copyright__ = "Copyright 2020, The SAGA Project"
+__license__   = "MIT"
+
+
+''' noop based job adaptor implementation '''
```
Comment (mturilli) [inline src/radical/saga/adaptors/noop/noop_job.py]: Should we use: no operation (noop)? Just trying to improve readibility but really non fundamental (who did not do some assembly in h[er|is] life?! :D

File: src/radical/saga/adaptors/noop/noop_job.py [hunk 8]
```diff
@@ -0,0 +1,683 @@
...
+                                 "class": "NoopJob"
+                         }]
+}
+
+
+# ------------------------------------------------------------------------------
+# The adaptor class
+class Adaptor(base.Base):
+    '''
+    This is the actual adaptor class, which gets loaded by SAGA (i.e. by the
```
Comment (mturilli) [inline src/radical/saga/adaptors/noop/noop_job.py]: i.e. -> i.e.,

File: src/radical/saga/adaptors/noop/noop_job.py [hunk 9]
```diff
@@ -0,0 +1,683 @@
...
+        return self.monitor.get_job(job_id)
+
+
+    # --------------------------------------------------------------------------
+    #
+    @SYNC_CALL
+    def container_run(self, jobs):
+        '''
+        From all the job descriptions in the container, build a bulk, and submit
+        as async.  The read whaterver the wrapper returns, and sort through the
```
Comment (mturilli) [inline src/radical/saga/adaptors/noop/noop_job.py]: The read -> Then read

File: src/radical/saga/adaptors/noop/noop_job.py [hunk 10]
```diff
@@ -0,0 +1,683 @@
...
+    #
+    @SYNC_CALL
+    def wait(self, timeout):
+        '''
+        A call to the noop to do the WAIT would block the noop for any
+        other interactions.  In particular, it would practically kill it if the
+        Wait waits forever...
+
+        So we implement the wait via a state pull.  The *real* solution is, of
+        course, to implement state notifications, and wait for such
```
Comment (mturilli) [inline src/radical/saga/adaptors/noop/noop_job.py]: I  would eliminate `, of course,`

File: docs/source/usage/config.rst (hunk 1 above)
Comment (andre-merzky) [reply docs/source/usage/config.rst]: Ack, see #760

File: docs/source/usage/logger.rst (hunk 2 above)
Comment (andre-merzky) [reply docs/source/usage/logger.rst]: Thanks, fixed (and some other items in that file corrected)

File: docs/source/usage/logger.rst (hunk 3 above)
Comment (andre-merzky) [reply docs/source/usage/logger.rst]: dito

File: docs/source/usage/logger.rst (hunk 4 above)
Comment (andre-merzky) [reply docs/source/usage/logger.rst]: dito

File: docs/source/usage/logger.rst (hunk 5 above)
Comment (andre-merzky) [reply docs/source/usage/logger.rst]: dito

File: examples/jobs/slurmjobcontainer.py (hunk 6 above)
Comment (andre-merzky) [reply examples/jobs/slurmjobcontainer.py]: thx, done.

File: src/radical/saga/adaptors/noop/noop_job.py (hunk 7 above)
Comment (andre-merzky) [reply src/radical/saga/adaptors/noop/noop_job.py]: hehe - done.

File: src/radical/saga/adaptors/noop/noop_job.py (hunk 9 above)
Comment (andre-merzky) [reply src/radical/saga/adaptors/noop/noop_job.py]: Thanks, that comment was off anyway - fixed.

File: src/radical/saga/adaptors/noop/noop_job.py (hunk 10 above)
Comment (andre-merzky) [reply src/radical/saga/adaptors/noop/noop_job.py]: done

**Comment by andre-merzky** [comment]
Thanks Matteo,  that helped!  I  hope I addressed all comments.

**Comment by mturilli** [comment]
All Travis tests fail. I would fix them before merging.

**Comment by andre-merzky** [comment]
The ssh context adaptor tests fail - but I can't see why from the Travis output, they work locally as expected :-(

**Pull request merged**